cmd/activator/main.go

```diff
@@ -241,19 +243,27 @@ func main() {
 	ah = &activatorhandler.ProbeHandler{NextHandler: ah}
 	ah = &activatorhandler.HealthHandler{HealthCheck: statSink.Status, NextHandler: ah}
 
+	profilingHandler := profiling.NewHandler(false)
 	// Watch the logging config map and dynamically update logging levels.
 	configMapWatcher.Watch(pkglogging.ConfigMapName(), pkglogging.UpdateLevelFromConfigMap(logger, atomicLevel, component))
-	// Watch the observability config map and dynamically update metrics exporter.
-	configMapWatcher.Watch(metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger))
-	// Watch the observability config map and dynamically update request logs.
-	configMapWatcher.Watch(metrics.ConfigMapName(), updateRequestLogFromConfigMap(logger, reqLogHandler))
+
+	// Watch the observability config map
+	for _, observer := range []func(*corev1.ConfigMap){
+		metrics.UpdateExporterFromConfigMap(component, logger),
+		updateRequestLogFromConfigMap(logger, reqLogHandler),
+		profiling.UpdateProfilingFromConfigMap(profilingHandler, logger),
+	} {
+		configMapWatcher.Watch(metrics.ConfigMapName(), observer)
+	}
```

Comment:
When I commented on this earlier, I kinda envisioned it to look more like this:

```golang
configMapWatcher.Watch(metrics.ConfigMapName(), func(cm *corev1.ConfigMap) {
	metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger)(cm)
	updateRequestLogFromConfigMap(logger, reqLogHandler)(cm)
})
```

The extra Watch calls might not actually hurt, to me they feel more expensive though. For posterity: I proposed to collapse these multiple Watch statements into one to cut down some repitition.

Reply:
Hmm. This is possible as well. I just don't know which of these solutions is more expensive. The original one registers more functions at the beginning (only once) but then calls them directly (possibly many times when a config map is changed during the lifetime of Knative). Your solution registers only one function at the beginning. And later calls this function indirectly multiple times (indirectly because it has to first call the wrappers which construct the function and then call the function itself). I don't have a strong opinion though.

Reply:
the one I posted has sliiiiightly more allocations (maybe?! don't know maybe it's even compiled away).

The point I'm after is calling `Watch` only once. It likely doesn't matter at all, performancewise. it's mostly for code hygiene.

Reply:
We could also refactor [this](https://github.com/knative/serving/blob/master/vendor/knative.dev/pkg/configmap/manual_watcher.go#L38) to accept `o ...Observer` and then call it like 
```
configMapWatcher.Watch(metrics.ConfigMapName(), metrics.ConfigMapName(), 
        metrics.UpdateExporterFromConfigMap(component, logger),
	updateRequestLogFromConfigMap(logger, reqLogHandler))
```

Reply:
that sounds like an even better idea to me! 👍